docs/pages/push-notifications/faq.mdx

```diff
@@ -61,61 +77,99 @@ curl -H "Content-Type: application/json" -X POST "https://exp.host/--/api/v2/pus
 
-And setting the priority to `high` gives your notification the greatest likelihood that the Android OS will display the notification.
+And setting the priority to `high` gives your notification the greatest likelihood that the Android will display the notification.
 
-### Does Expo store the contents of push notifications?
+### Handle expired push notification credentials
 
-Expo does not store the contents of push notifications any longer than it takes to deliver the notifications to the push notification services operated by Apple, Google, etc... Push notifications are stored only in memory and in message queues and **not** stored in databases.
+When your push notification credentials have expired, run `eas credentials`, choose iOS and a build profile, then remove your push notification key and generate a new one.
 
-### Does Expo read or share the contents of push notifications?
+### Error message when sending a notification
 
-Expo does not read or share the contents of push notifications and our services keep push notifications only as long as needed to deliver them to push notification services run by Apple and Google. If the Expo team is actively debugging the push notifications service, we may see notification contents (ex: at a breakpoint) but Expo cannot see push notification contents otherwise.
+Check the `details` property of the returned push ticket or receipt for more information that you can use to debug further. [Read here for common error code responses and their associated solutions](/push-notifications/sending-notifications/#errors).
 
-### How does Expo encrypt connections to push notification services, like Apple's and Google's?
+### Push token fetched takes a long time on iOS
 
-Expo's connections to Apple and Google are encrypted and use HTTPS.
+`getDevicePushTokenAsync` and `getExpoPushTokenAsync` can sometimes take a long time to resolve on iOS. This is outside of `expo-notifications`'s control, as stated in Apple's [Troubleshooting Push Notifications](https://developer.apple.com/library/archive/technotes/tn2265/_index.html) technical note:
 
-### How do I handle expired push notification credentials?
+> This is not necessarily an error condition. The system may not have Internet connectivity at all because it is out of range of any cell towers or Wi-Fi access points, or it may be in airplane mode. Instead of treating this as an error, your app should continue normally, disabling only that functionality that relies on push notifications.
 
-When your push notification credentials have expired, run `eas credentials`, choose iOS and a build profile, then remove your push notification key and generate a new one.
+As mentioned, the most common reasons for this issue are either an invalid Internet connection (fetching a push token requires an Internet connection to register the device with the service provider) or an invalid configuration of your App ID or Provisioning Profile.
```

Comment:
```suggestion
```
This paragraph is well written but it re-states the quote from Apple directly, which I don't think is needed.